### PR TITLE
[POU-70] Search bar with search

### PR DESCRIPTION
**Pull request by MingLiang-PALOIT**
JIRA Ticket: POU-70

Description (Optional):
Implemented Search bar with search

### CODE REVIEW

File: src/app/search/projects/page.tsx [hunk 1]
```diff
@@ -1,29 +1,35 @@
...
-import { NextPage } from 'next'
 import { useEffect, useLayoutEffect } from 'react'
-import { Container, Flex, Box, Button, Hide } from '@chakra-ui/react'
-import { shouldReposition, setScroll } from '@/utils/Stickify'
+import { NextPage } from 'next'
+import { useSearchParams } from 'next/navigation'
+import { Box, Button, Container, Flex, Hide } from '@chakra-ui/react'
+
+import { useActions, useAppState, useEffects } from '@/overmind'
+import { ALLOWED_RENDER_TYPE } from '@/@types/ProjectSearchResult'
```
Comment (morphet81) [inline src/app/search/projects/page.tsx:8]: What is ALLOWED_RENDER_TYPE?

File: src/app/search/projects/page.tsx [hunk 2]
```diff
@@ -1,29 +1,35 @@
...
+import { setScroll, shouldReposition } from '@/utils/Stickify'
+
 import { ProjectSearchHead } from '@/components/molecules/ProjectSearchHead/ProjectSearchHead'
 import { ProjectSearchBody } from '@/components/molecules/ProjectSearchBody/ProjectSearchBody'
 import { PaginationWidget } from '@/components/atoms/PaginationWidget/PaginationWidget'
-import { useActions, useEffects } from '@/overmind'
 import { DownloadIcon } from '@/components/atoms/DownloadIcon/DownloadIcon'
-import { ALLOWED_RENDER_TYPE } from '@/@types/ProjectSearchResult'
 
 const DEFAULT_PROJECT_TO_DISPLAY = 15
```
Comment (morphet81) [inline src/app/search/projects/page.tsx:16]: What's this?

File: src/components/molecules/SearchHeader/SearchHeader.tsx [hunk 3]
```diff
@@ -1,13 +1,34 @@
...
+import { BreadCrumbs } from '@/components/atoms/BreadCrumbs/BreadCrumbs'
 
 export const SearchHeader = () => {
+  const [searchInput, setSearchInput] = useState(``)
   const currentPath = usePathname()
   const currentTitle = extractTitleFromUrl(currentPath)
   const { t } = useTranslation(`search`)
+  const searchParams = useSearchParams()
+  const router = useRouter()
+  const pattern = searchParams.get(`keyword`) ?? ``
```
Comment (morphet81) [inline src/components/molecules/SearchHeader/SearchHeader.tsx:17]: String like params names can be put in enums for better reusability.

Comment (morphet81) [reply src/components/molecules/SearchHeader/SearchHeader.tsx:17]: And change it everywhere it's used.

File: src/overmind/projectResult/effects.ts [hunk 4]
```diff
@@ -1,61 +1,39 @@
...
-    //   })
-    //   .finally(() => {
-    //         resolve(result)
-    //   })
+    let result: EffectResponse<ProjectSearchResponse>
+
+    const searchParams = new URLSearchParams()
+    searchParams.append(`pattern`, pattern)
+    searchParams.append(`count`, count.toString())
+    searchParams.append(`offset`, offset.toString())
```
Comment (morphet81) [inline src/overmind/projectResult/effects.ts:16]: Params in enum

File: src/overmind/projectResult/effects.ts [hunk 5]
```diff
@@ -1,61 +1,39 @@
...
-import { DateFormats } from '@/@types/DateFormats'
 import { EffectResponse } from '@/@types/EffectResponse'
-import { ProjectSearchResult } from '@/@types/ProjectSearchResult'
-import { projectData } from '@/test/mock-data/projects_data'
-import { formatDate } from '@/utils/DateFormat'
+import { ProjectSearchResponse } from '@/@types/ProjectSearchResult'
 import { ProjectDetails } from '@/@types/ProjectDetails'
 
-const SLEEP = 3000
+const DEFAULT_COUNT = 15
```
Comment (morphet81) [inline src/overmind/projectResult/effects.ts:7]: Naming can be better.

File: src/styles/Section.ts [hunk 6]
```diff
@@ -32,3 +32,16 @@ export const filterCardSection = defineStyle({
...
+export const searchBar = defineStyle({
+  color: `black`,
+  boxShadow: `2px 2px 8px 0px #0000001A;`,
+  minHeight: `min-content`,
+  borderRadius: `12px`,
+  background: `#FFFFFF`,
+  margin: 0,
+  padding: `16px`,
+  maxW: `unset`,
+
```
Comment (morphet81) [inline src/styles/Section.ts:45]: extra line

**Review by morphet81 — CHANGES_REQUESTED** [review]

File: src/app/search/projects/page.tsx (hunk 1 above)
Comment (MingLiang-PALOIT) [reply src/app/search/projects/page.tsx:8]: ALLOWED_RENDER_TYPE is used to throw an error if the table with the "stickify" scrolling is rendered in a component that is not allowed. This stickify functionality is used predominantly in the project search result for now and I believe it was put in place as after a discussion be you and Mederic previously

File: src/app/search/projects/page.tsx [hunk 7]
```diff
@@ -82,7 +88,11 @@ const ProjectPage: NextPage = () => {
       <ProjectSearchHead renderType={ALLOWED_RENDER_TYPE} />
       <ProjectSearchBody renderType={ALLOWED_RENDER_TYPE} />
       <Container variant={`paginationBar`}>
-        <PaginationWidget onPageChange={(currentPage, from) => getNewResults(from)} resultPerPage={DEFAULT_PROJECT_TO_DISPLAY} totalResults={89} />
+        <PaginationWidget
+          onPageChange={(currentPage, from) => getNewResults(from)}
```
Comment (morphet81) [inline src/app/search/projects/page.tsx:92]: No inline

**Review by morphet81 — CHANGES_REQUESTED** [review]

**Review by morphet81 — APPROVED** [review]

**Pull request merged**